resources/process-openmaptiles.lua

```diff
@@ -405,117 +441,156 @@ function way_function()
+		if linkValues[h] then
 			splitHighway = split(highway, "_")
 			highway = splitHighway[1]; h = highway
 			ramp = true
-			minzoom = 11
 		end
 
 		-- Construction
-		if highway == "construction" then
-			if constructionValues[construction] then
-				h = construction .. "_construction"
-				if construction ~= "service" and construction ~= "track" then
-					minzoom = 11
-				else
-					minzoom = 12
-				end
-			else
-				h = "minor_construction"
-				minzoom = 14
-			end
+		if under_construction then
+			h = h .. "_construction"
+		end
+
+		-- Drop underground platforms
+		local layer = Find("layer")
+		local layerNumeric = tonumber(layer)
+		if not is_road and layerNumeric ~= nil and layerNumeric < 0 then
+			minzoom = INVALID_ZOOM
+		end
+
+		-- Drop all areas except infrastructure for pedestrians handled above
+		if isClosed and h ~= "path" then
+			minzoom = INVALID_ZOOM
 		end
 
 		-- Write to layer
 		if minzoom <= 14 then
-			write_to_transportation_layer(minzoom, h)
+			write_to_transportation_layer(minzoom, h, subclass, ramp, service, false, is_road)
 
 			-- Write names
-			if minzoom < 8 then
-				minzoom = 8
-			end
-			if highway == "motorway" or highway == "trunk" then
+			if not isClosed and (HasNames() or Holds("ref")) then
+				if h == "motorway" then
+					minzoom = 7
+				elseif h == "trunk" then
+					minzoom = 8
+				elseif h == "primary" then
+					minzoom = 10
+				elseif h == "secondary" then
+					minzoom = 11
+				elseif h == "minor" or h == "track" or h == "tertiary" then
+					minzoom = 13
+				else
+					minzoom = 14
+				end
 				Layer("transportation_name", false)
 				MinZoom(minzoom)
-			elseif h == "minor" or h == "track" or h == "path" or h == "service" then
-				Layer("transportation_name_detail", false)
-				MinZoom(minzoom)
-			else
-				Layer("transportation_name_mid", false)
-				MinZoom(minzoom)
-			end
-			SetNameAttributes()
-			Attribute("class",h)
-			Attribute("network","road") -- **** could also be us-interstate, us-highway, us-state
-			if h~=highway then Attribute("subclass",highway) end
-			local ref = Find("ref")
-			if ref~="" then
-				Attribute("ref",ref)
-				AttributeNumeric("ref_length",ref:len())
+				SetNameAttributes()
+				Attribute("class",h)
+				Attribute("network","road") -- **** could also be us-interstate, us-highway, us-state
+				if subclass then Attribute("subclass", highway) end
+				local ref = Find("ref")
+				if ref~="" then
+					Attribute("ref",ref)
+					AttributeNumeric("ref_length",ref:len())
+				end
 			end
 		end
 	end
 
-	-- Railways ('transportation' and 'transportation_name', plus 'transportation_name_detail')
+	-- Railways ('transportation' and 'transportation_name')
 	if railway~="" then
-		Layer("transportation", false)
-		Attribute("class", railway)
-		SetZOrder()
-		SetBrunnelAttributes()
-		if service~="" then
-			Attribute("service", service)
-			MinZoom(12)
-		else
-			MinZoom(9)
-		end
+		local class = railwayClasses[railway]
```

Comment:
I noticed that railway=proposed also ends up in the tiles. Am I right in thinking that this now takes care of it?

Reply:
Neither the schema documentation nor the official implementation nor the tiles served by MapTiler contain `railway=proposed`.

`railwayClasses` contains all accepted values of `railway=*`.

Reply:
That was my point. It ended up in the tiles before this PR and presumably doesn't anymore, correct?

Reply:
Yes, I will not end up in the tiles any more.